Automattic/newspack-theme
# Refactor: Simplify the mobile menu files

### All Submissions:


### Changes proposed in this Pull Request:

In anticipation of potentially adding more menus (#766), this PR simplifies the mobile menu code. Rather than splitting it over two files (AMP and non-AMP), it combines them into one with some conditional checks before outputting the AMP-specific code. There's enough of a significant code overlap, with potentially more to come, that this seems easier to manage. 

### How to test the changes in this Pull Request:

1. Apply the PR.
2. Make sure you don't have a short header set, and that you have menus assigned to the four locations in the header (Primary, Secondary, Tertiary and Social).
3. With AMP on, confirm all these menus display as expected.
4. Shrink down the browser to a mobile-sized screen, and confirm that the menu slideout works.
5. Turn off AMP.
6. Retest the mobile menu and confirm it still works. 

### Other information:
